sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestController.java

```diff
@@ -201,52 +203,43 @@ private void handleAssociatedCase(
 		// a.2) Case classification != confirmed: Ask user whether to confirm the case
 		// b) Tested disease != case disease: Ask user to create a new case for the tested disease
 
-		CaseDataDto preSaveCaseDto = FacadeProvider.getCaseFacade().getCaseDataByUuid(associatedCase.getUuid());
-		CaseDataDto postSaveCaseDto = FacadeProvider.getCaseFacade().getCaseDataByUuid(associatedCase.getUuid());
-		showSaveNotification(preSaveCaseDto, postSaveCaseDto);
+		CaseDataDto caze = FacadeProvider.getCaseFacade().getCaseDataByUuid(associatedCase.getUuid());
 
-		final boolean equalDisease = dto.getTestedDisease() == postSaveCaseDto.getDisease();
+		final boolean equalDisease = dto.getTestedDisease() == caze.getDisease();
 
-		Runnable negativeTestCallback = () -> {
+		Runnable callback = () -> {
 			if (equalDisease && PathogenTestResultType.NEGATIVE.equals(dto.getTestResult()) && dto.getTestResultVerified()) {
-				changeAssociatedSampleResult(dto, handleChanges -> {
+				showChangeAssociatedSampleResultDialog(dto, handleChanges -> {
 				});
 			}
-		};
 
-		Runnable positiveTestCallback = () -> {
 			if (PathogenTestResultType.POSITIVE.equals(dto.getTestResult()) && dto.getTestResultVerified()) {
```

Comment:
Only minor, but for better readability (complexity makes this code hard to read anyways): `else if`